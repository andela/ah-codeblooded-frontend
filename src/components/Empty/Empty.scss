.empty {
  padding: 4em;
}

.icon {
  font-size: 4em !important;
}

.message {
  white-space: nowrap;
}
