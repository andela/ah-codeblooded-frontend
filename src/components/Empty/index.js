import React from 'react';
import PropTypes from 'prop-types';
import './Empty.scss';

const Empty = ({ message, icon }) => (
  <div className="card empty">
    <div className="card-content">
      <div className="row">
        <div className="center col s12 m6 offset-m3">
          <i className="material-icons icon grey-text">{ icon }</i>
          <div className="message">
            { message }
          </div>
        </div>
      </div>
    </div>
  </div>
);

Empty.defaultProps = {
  icon: 'hourglass_empty',
  message: 'No Items',
};

Empty.propTypes = {
  message: PropTypes.string,
  icon: PropTypes.string,
};

export default Empty;
