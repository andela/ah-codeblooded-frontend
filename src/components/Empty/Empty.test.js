import React from 'react';
import { shallow } from 'enzyme';
import Empty from './index';

describe('<Empty> component', () => {
  it('should render without crushing', () => {
    const wrapper = shallow(<Empty />);
    expect(wrapper).toMatchSnapshot();
  });
});
