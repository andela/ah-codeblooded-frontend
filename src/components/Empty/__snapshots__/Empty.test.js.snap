// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty> component should render without crushing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Empty
    icon="hourglass_empty"
    message="No Items"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-content"
      >
        <div
          className="row"
        >
          <div
            className="center col s12 m6 offset-m3"
          >
            <i
              className="material-icons icon grey-text"
            >
              hourglass_empty
            </i>
            <div
              className="message"
            >
              No Items
            </div>
          </div>
        </div>
      </div>,
      "className": "card empty",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="center col s12 m6 offset-m3"
          >
            <i
              className="material-icons icon grey-text"
            >
              hourglass_empty
            </i>
            <div
              className="message"
            >
              No Items
            </div>
          </div>
        </div>,
        "className": "card-content",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="center col s12 m6 offset-m3"
          >
            <i
              className="material-icons icon grey-text"
            >
              hourglass_empty
            </i>
            <div
              className="message"
            >
              No Items
            </div>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="material-icons icon grey-text"
              >
                hourglass_empty
              </i>,
              <div
                className="message"
              >
                No Items
              </div>,
            ],
            "className": "center col s12 m6 offset-m3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "hourglass_empty",
                "className": "material-icons icon grey-text",
              },
              "ref": null,
              "rendered": "hourglass_empty",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No Items",
                "className": "message",
              },
              "ref": null,
              "rendered": "No Items",
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-content"
        >
          <div
            className="row"
          >
            <div
              className="center col s12 m6 offset-m3"
            >
              <i
                className="material-icons icon grey-text"
              >
                hourglass_empty
              </i>
              <div
                className="message"
              >
                No Items
              </div>
            </div>
          </div>
        </div>,
        "className": "card empty",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="center col s12 m6 offset-m3"
            >
              <i
                className="material-icons icon grey-text"
              >
                hourglass_empty
              </i>
              <div
                className="message"
              >
                No Items
              </div>
            </div>
          </div>,
          "className": "card-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="center col s12 m6 offset-m3"
            >
              <i
                className="material-icons icon grey-text"
              >
                hourglass_empty
              </i>
              <div
                className="message"
              >
                No Items
              </div>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="material-icons icon grey-text"
                >
                  hourglass_empty
                </i>,
                <div
                  className="message"
                >
                  No Items
                </div>,
              ],
              "className": "center col s12 m6 offset-m3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "hourglass_empty",
                  "className": "material-icons icon grey-text",
                },
                "ref": null,
                "rendered": "hourglass_empty",
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "No Items",
                  "className": "message",
                },
                "ref": null,
                "rendered": "No Items",
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
